.github/actions/go-cache-push-s3/action.yml

```diff
@@ -0,0 +1,93 @@
+name: 'Save Go essentials into S3'
+description: 'Push the Go essentials into S3'
+inputs:
+  key:
+    description: Cache key
+    required: true
+  paths:
+    description: Where files/folders is located
+    required: true
+  aws-s3-bucket:
+    description: Bucket to save cache
+    required: true
+  aws-access-key-id:
+    description: Access key to AWS
+    required: true
+  aws-secret-access-key:
+    description: Secret access key to AWS
+    required: true
+  override:
+    description: Always override the cache
+    required: false
+    default: "true"
+runs:
+  using: "composite"
+  steps:
+    - name: Do not push cache when exists
+      shell: bash
+      run: |
+        if '${{ inputs.override }}' == 'true'; then
+          echo "SKIP=false" >>  $GITHUB_ENV
+          exit 0
+        fi
+
+        key=$(aws s3 ls s3://${{ inputs.aws-s3-bucket }} | tr -s ' ' | cut -d ' ' -f 4 | grep '${{ runner.os }}-go-${{ env.GO_VERSION }}-v6-${{ inputs.key }}-${{ hashFiles('**/go.sum') }}') || true
+        echo "SKIP=false" >> $GITHUB_ENV
+
+      env:
+        AWS_ACCESS_KEY_ID: ${{ inputs.aws-access-key-id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.aws-secret-access-key }}
+        AWS_DEFAULT_REGION: 'us-east-1'
+
+    - name: Convert multiline string to singleline
+      shell: python
```

Comment:
I understand, The task is to check whether the given `paths` exists in direcotry structure. At first I wanted to use `stat` however it ends up with an error when the `directory/file` does not exists. And in CI it is very not good solution. When shell ends up with status code different than `0`, it stops. Therefore I have to use already for `aws cli` everywhere `key=$(cmd 2&>1) || true` to suppress the status code and actually check the stdout/stderr.
I came accross this solution which I found more convinient due to no need of suppressing error and just plain iteration over paths that exists, rest of paths are concatenated into final path list that is compressed.
If I would want to convert everyting into python It would be more boilerplate than this nested statements in `shell`.